frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -233,7 +233,7 @@ function QueryBuilder(props) {
   const wasShowingAnySidebar = usePrevious(isAnySidebarOpen);
   const wasNativeEditorOpen = usePrevious(isNativeEditorOpen);
   const hasQuestion = question != null;
-  const collectionId = question?.collectionId();
+  const collectionId = card?.collectionId || question?.collectionId();
```

Comment:
If you have to do this something went wrong. We shouldn't be using `card` directly anywhere